woocommerce/woocommerce-ios
# Remove `.hubMenu` feature flag references

### Description:
As part of https://github.com/woocommerce/woocommerce-ios/issues/7605 (removing feature-flag-related code that has been in production for a while) this PR removes the `.hubMenu` feature flag, as well as logic that relies on it.

### Changes:
- Removes the `.hubMenu` feature flag
- Removes the `DashBoardViewController` navigation button fallback when `hubMenu` is disabled, this added a "gear" icon to the My Store screen, which is no longer necessary.
- Removes the `SettingsViewModel` additional row from existing sections when `hubMenu` is disabled, is no longer needed as the HubMenu always shows up.
- `MainTabBarController` is no longer computed, always returns the default 4 WooTabs that are expected at the bottom of the app: `[.myStore, .orders, .products, .hubMenu]`.
- Removes the Reviews tab coordinator configuration, that acts as a fallback when the `hubMenu` is disabled, as the HubMenu always shows up. 
- Removes the `hubMenu` feature flag from `MockFeatureFlagService`, which is no longer needed to filter unit test behaviour, as well as edited the tests as needed.

### Testing instructions:
1. Navigate to "My Store" and confirm that no gear icon appears in the top right corner.
2. Navigate to "Menu" and confirm that brings the Hub Menu if tapped.
3. Confirm that Unit Tests pass.

<img width="400" alt="Screenshot 2022-09-01 at 19 22 37" src="https://user-images.githubusercontent.com/3812076/187975125-5293fdd4-d315-4b77-ad6d-60f4a8fb2b3b.png">


---
